ramips: Add support for D-Link DIR-2150-A1

Lucky1openwrt opened the pull request:
Hardware Specification:
SoC: Mediatek MT7621DAT (MIPS1004Kc 880 MHz, dual core)
RAM: 128 MB
Storage: 128 MB NAND flash
Ethernet: 5x 10/100/1000 Mbps LAN1,LAN2,LAN3,LAN4 & WAN 
Wireless: 2.4GHz: Mediatek MT7603EN up to 300Mbps (802.11b/g/n MIMO 2x2) 
Wireless: 5GHz: Mediatek MT7615N up to 1733Mbps (802.11n/ac MU-MIMO 4x4) 
LEDs: Power (white & amber), Internet (white & amber) 
LEDs: 2.4G (White), 5Ghz (White)
Buttons: WPS, Reset

MAC Table
Label	xx:xx:xx:xx:xx:EB
LAN         xx:xx:xx:xx:xx:EB
2.4Ghz	xx:xx:xx:xx:xx:EC
5Ghz	xx:xx:xx:xx:xx:ED
WAN	xx:xx:xx:xx:xx:EE

Flash instructions:
D-Link normal OEM firmware update page
	1. upload OpenWRT factory.bin like any D-Link upgrade image

D-Link Recovery GUI:
    1. Push and hold reset button (on the bottom of the device) until power led starts flashing (about 10 secs or so) while plugging in the power cable.
    2. Give it ~30 seconds, to boot the recovery mode GUI
    3. Connect your client computer to LAN1 of the device
    4. Set your client IP address manually to 192.168.0.2 / 255.255.255.0
    5. Call the recovery page for the device at http://192.168.0.1/
    6. Use the provided emergency web GUI to upload the recovery.bin to the device

Firefox on Windows in a Private Window (incognito) works me
Internet Explorer mode in Microsoft Edge works for others
seems to not work in Linux or virtual machine on Linux for most
some see success using 'curl -v -i -F "firmware=@file.bin" 192.168.0.1'

Thanks to @frkca and @rodneyrod for testing and pushing for its creation

Successor of PR https://github.com/openwrt/openwrt/pull/15275
update of https://github.com/openwrt/openwrt/pull/12624

Lucky1openwrt commented:
update of https://github.com/openwrt/openwrt/pull/12624
as there is no response from @frkca

Lucky1openwrt commented:
@rodneyrod
things to double check
mapping of MAC address in D-Link OEM firmware & case label ?
mapping and correct operation of WIFI leds ? 
conformation that the firmware.bin image will upgrade correctly via D-Link 's normal wed interface
conformation that the recovery.bin image will upload via the emergence interface
ram confirmation as 128M as the other models with USB have 256M ? 
http://luckys.onmypc.net/openwrt/DIR-2150-A1/

namiltd commented:
The mt7621_dlink_dir-xx60-a1.dtsi file is loaded in
mt7621_dlink_dir-1960-a1.dts
mt7621_dlink_dir-2640-a1.dts
mt7621_dlink_dir-2660-a1.dts
mt7621_dlink_dir-3040-a1.dts
and now additionally in mt7621_dlink_dir-2150-a1.dts
Maybe change its name to something common, e.g. mt7621_dlink_dir-xxx0-a1.dts ?

Lucky1openwrt commented:
true not much xx60 left
next update maybe i'll call it
mt7621_dlink_dir-NAND-a1.dtsi
or is that t generic ?

namiltd commented:
I propose mt7621_dlink_dir_nand_128m.dtsi

Lucky1openwrt commented:
sounds good :) as it's really -Ax as well -A1 & -A2

namiltd commented:
A small note: it should be compatible with other files, e.g. mt7621_xiaomi_nand_128m.dtsi
should be mt7621_dlink_dir_nand_128m.dtsi
and not mt7621_dlink_dir-nand_128m.dtsi

namiltd commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -769,6 +769,16 @@ define Device/dlink_dir-1960-a1
 TARGET_DEVICES += dlink_dir-1960-a1
 
+define Device/dlink_dir-2150-a1
+  $(Device/dlink_dir-xx60-a1)
```
this can also be changed to the new name and in appropriate other places

namiltd commented:
You can also make a mess with the nomenclature
mt7621_dlink_dir-2150-a1.dts:
```
	leds: leds {
		compatible = "gpio-leds";

		led_power_orange: led-0 {
			function = LED_FUNCTION_POWER;
			color = <LED_COLOR_ID_ORANGE>;
			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
		};

		led_power_white: led-1 {
			function = LED_FUNCTION_POWER;
			color = <LED_COLOR_ID_WHITE>;
			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
		};

		led_net_orange: led-2 {
			function = LED_FUNCTION_WAN;
			color = <LED_COLOR_ID_ORANGE>;
			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
		};

		led-3 {
			function = LED_FUNCTION_WAN;
			color = <LED_COLOR_ID_WHITE>;
			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
		};
	};
```
01_leds:
```
dlink,dir-1960-a1|\
dlink,dir-2150-a1|\
dlink,dir-2640-a1|\
dlink,dir-2660-a1)
	ucidef_set_led_netdev "wan" "wan" "white:wan" "wan"
```
You can also look at the remaining files based on mt7621_dlink_dir_nand_128m.dtsi

namiltd commented:
I did it myself: https://github.com/openwrt/openwrt/pull/15275

namiltd commented:
Please do:
```
git rebase 3a55c8
```
and then your changes.

namiltd commented:
It would make sense if your patch came after mine.

You can do it e.g. like this:
```
git reset --hard 84a48c

git remote add other https://github.com/namiltd/openwrt.git
git fetch other
git cherry-pick 8b687a
git cherry-pick 3a55c8
git remote remove other
```
and then your patch (updated to new names)

Lucky1openwrt commented:
its ok it's i think it will be after 
it's deliberate in draft awaiting confirmations from @rodneyrod
I'll update it soon tho with the changes tho
I don't have this exact device only it's bother so most tings tested
just need some t's crossed & i's dotted

namiltd commented:
> its ok it's i think it will be after

Fixed. Sync to my tree again (I updated the hashes)

rodneyrod commented:
Update on the test results, just able to compare the results from the previous builds you put up (thanks again!) and the restore of the stock firmware from the unencrypted .bin you provided.
When everyone who needs the net now are off tonight, I'll be able to test installing OpenWRT from the stock firmware.

>mapping of MAC address in D-Link OEM firmware & case label ?
MAC matches

>mapping and correct operation of WIFI leds ?
Yes

>conformation that the firmware.bin image will upgrade correctly via D-Link 's normal wed interface
TBD

>conformation that the recovery.bin image will upload via the emergence interface
Yes

>ram confirmation as 128M ?
Yes

>No USB confirmation ?
Confirmed, USB options not listed in LuCI

Just of note, didn't have much luck with Firefox on Linux or Windows flashing from recovery but falling back to Edge in IE mode is confirmed working.

Lucky1openwrt commented:
thanks 
so still need to test the basic firmware.bin update via normal D-Link firmware update interface
there is a certificate involved that needs testing or I'll remove the creation of the firmware file 
 
about the USB the Unit has no physical usb ports available rear of front ?

Lucky1openwrt committed:
ramips: Add support for D-Link DIR-2150-A1

Hardware Specification:
SoC: Mediatek MT7621DAT (MIPS1004Kc 880 MHz, dual core)
RAM: 128 MB
Storage: 128 MB NAND flash
Ethernet: 5x 10/100/1000 Mbps LAN1,LAN2,LAN3,LAN4 & WAN
Wireless: 2.4GHz: Mediatek MT7603EN up to 300Mbps (802.11b/g/n MIMO 2x2)
Wireless: 5GHz: Mediatek MT7615N up to 1733Mbps (802.11n/ac MU-MIMO 4x4)
LEDs: Power (white & amber), Internet (white & amber)
LEDs: 2.4G (White), 5Ghz (White)
Buttons: WPS, Reset

MAC Table
Label xx:xx:xx:xx:xx:EB
LAN xx:xx:xx:xx:xx:EB
2.4Ghz xx:xx:xx:xx:xx:EC
5Ghz xx:xx:xx:xx:xx:ED
WAN xx:xx:xx:xx:xx:EE

Flash instructions:
D-Link normal OEM firmware update page
1. upload OpenWRT factory.bin like any D-Link upgrade image

D-Link Recovery GUI:
1. Push and hold reset button (on the bottom of the device) until power led starts flashing (about 10 secs or so) while plugging in the power cable.
2. Give it ~30 seconds, to boot the recovery mode GUI
3. Connect your client computer to LAN1 of the device
4. Set your client IP address manually to 192.168.0.2 / 255.255.255.0
5. Call the recovery page for the device at http://192.168.0.1/
6. Use the provided emergency web GUI to upload the recovery.bin to the device

Firefox on Windows in a Private Window (incognito) works me
Internet Explorer mode in Microsoft Edge works for others
seems to not work in Linux or virtual machine on Linux for most
some see success using 'curl -v -i -F "firmware=@file.bin" 192.168.0.1'

Thanks to @frkca and @rodneyrod for testing and pushing for its creation

Signed-off-by: Alan Luck <luckyhome2008@gmail.com>

Lucky1openwrt commented:
foh well something broke this how to unclose ?

namiltd commented:
You forgot "ramips:" on commit title.

rodneyrod commented:
Got my final updates done, install from factory confirmed working, was having some issues with 160Mhz 802.11ac but that's now working fine  (needed me to change the channel to 36, found this after a bit of googling, apparently 100 will work too but haven't tested this)

 No physical USB ports on the device, there may be internal headers but haven't opened my unit to check this.

Was even able to import my old config from the previous test images you made and that's all working fine too.

Lucky1openwrt commented:
thanks all sounds good
the other USB models have 256M of ram +5v regulators and connectoer there is a mit missing if you where to add them
all good I'll take it out of draft :)

Lucky1openwrt commented:
@zkabelac @raenye @csharper2005
Hi guys as you have reviewed the old DIR-2150-A1 pull request
maybe you can review and comment on this one to help move it forward

namiltd commented:
PR https://github.com/openwrt/openwrt/pull/15275 has been updated. I also patched several Archer TP-Link routers.

csharper2005 commented on target/linux/ramips/mt7621/base-files/etc/uci-defaults/04_led_migration:
```diff
@@ -8,6 +8,13 @@ tplink,archer-a6-v3|\
 tplink,archer-c6-v3)
 	migrate_leds ':wifi2g$=:wlan-2' ':wifi5g$=:wlan-5'
 	;;
+dlink,dir-1960-a1|\
+dlink,dir-2640-a1|\
+dlink,dir-2660-a1|\
+dlink,dir-3040-a1|\
+dlink,dir-3060-a1)
+	migrate_leds ':net=:wan'
+	;;
```
Please sort this block alphabetically.

namiltd commented:
PR https://github.com/openwrt/openwrt/pull/15275 has been updated. I also patched several Archer TP-Link routers.

Lucky1openwrt commented:
all updated Thanks

namiltd commented:
My patches have been approved, rebase your PR.
